content/posts/aws-workplace.md

```diff
@@ -0,0 +1,96 @@
+
+Chmura Google rozwiązuje ten problem (przynajmniej częściowo) poprzez swój [Cloud Shell](https://cloud.google.com/shell) - darmowy terminal w przeglądarce, w którym można skorzystać z zawsze aktualnych narzędzi konsolowych do pracy z GCP. W przypadku AWS nie jest już tak prosto i - choć da się stworzyć podobne narzędzie samemu (np. instancja EC2 z przypisaną odpowiednią rolą IAM) - to będzie to wymagało dodatkowej pracy, a użyte zasoby będą płatne. Dlatego do celów szkoleniowo-testowych zdecydowałem się na pracę z AWS z konsoli laptopa.
+
+W trakcie zbierania informacji na temat tego jak skonfigurować swoje środowisko robocze do AWS trafiłem na kilka rzeczy, które chciałbym zrobić inaczej niż w podstawowej dokumentacji. Po pierwsze, domyślny sposób przechowywania kluczy dostępowych do AWS to niezaszyfrowany plik w katalogu domowym: komenda `aws configure` zapisuje je do `~/.aws/credentials`. Nawet jeżeli korzystamy z wieloetapowego uwierzytelniania (co jest rekomendowaną metodą dla użytkowników-ludzi), trzymanie niezabezpieczonych kluczy na dysku to nienajlepszy pomysł. Po drugie, oficjalna metoda instalacji narzędzia `aws-cli` to ściągnięcie paczki zip i uruchomienie skryptu instalacyjnego przy użyciu sudo[^1] - skrypt kopiuje sporą liczbę plików do globalnych folderów systemowych. Po trzecie, wieloetapowe uwierzytelnianie kodem OTP w konsoli wymaga wykonania dość długiej komendy i żonglerki kluczami dostępowymi, co warto trochę zautomatyzować. Po zlokalizowaniu problemów do pokonania i przeczytaniu kilku artykułów na ten temat (spis znajdziesz na końcu artykułu) postanowiłem zebrać najlepsze cechy znalezionych rozwiązań i napisać własny prosty skrypt, który będzie moim codziennym kompanem w pracy z chmurą Amazona.
+
+## Dla niecierpliwych
+
+Klonujemy [repozytorium git ze skryptami](https://github.com/ninefiveslabs/aws-workplace):
+```
+git clone https://github.com/ninefiveslabs/aws-workplace.git
+cd aws-workplace
+```
+
+Uruchamiamy skrypt `init.sh`, który zaszyfruje klucze:
+```
+./init.sh
+```
+Skrypt poprosi kolejno o: access key ID, secret access key, identyfikator urządzenia MFA (do znalezienia na podanej stronie, lub do założenia w [ten sposób](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_mfa_enable_virtual.html)), domyślny region i hasło do zaszyfrowania kluczy w pliku.
+```
+Enter your AWS_ACCESS_KEY_ID:
+Enter your AWS_SECRET_ACCESS_KEY:
+Enter your MFA device ARN (see: https://console.aws.amazon.com/iam/home#/security_credentials):
+Enter default region [us-east-1]:
+Now comes the GPG encryption password prompt:
+Enter passphrase:
+```
+
+Teraz za każdym razem, gdy będziemy chcieli przygotować konsolę do pracy z AWS, należy wczytać plik `env.sh`, który poprosi o podanie hasła do rozszyfrowania kluczy dostępowych oraz aktualnego kodu MFA:
```

Comment:
Moze wspomnieć o autoenv? https://github.com/inishchith/autoenv

Reply:
Fajne, albo nawet https://github.com/direnv/direnv. Używasz tego?

Reply:
Jak na razie autoenv spelnia moze potrzeby i nie czuje potrzeby zmiany, glownie dlatego, ze nie   mam potrzeby uzywania dodatkowych funkcjonalnosci jakie ma direnv.
Co do 'unload env vars' nie jest to dla mnie problemem bo z przywyczajenia odpala nowy terminal per projekt.

https://github.com/direnv/direnv/issues/159